src/bluetoothdevice.cpp

```diff
@@ -278,11 +286,40 @@ QStringList bluetoothdevice::metrics() {
 
 resistance_t bluetoothdevice::maxResistance() { return 100; }
 
+bool bluetoothdevice::updateLockscreenHeartRate() {
+
+#ifdef Q_OS_IOS
+#ifndef IO_UNDER_QT
+    lockscreen h;
+    long appleWatchHeartRate = h.heartRate();
+    this->Heart = appleWatchHeartRate;
+    qDebug() << "Current Heart from Apple Watch: " << QString::number(appleWatchHeartRate);
+    return true;
+#endif
+#endif
+    return false;
+}
+
+bool bluetoothdevice::updateLockscreenEnergyDistanceHeartRate(long defaultHeartRate) {
+#ifdef Q_OS_IOS
+#ifndef IO_UNDER_QT
+    lockscreen h;
+    long appleWatchHeartRate = h.heartRate();
+    h.setKcal(KCal.value());
+    h.setDistance(Distance.value());
+    this->Heart = appleWatchHeartRate!=0 ? appleWatchHeartRate:defaultHeartRate;
+    qDebug() << "Current Heart from Apple Watch: " << QString::number(appleWatchHeartRate);
```

Comment:
Changed from using the subclass' debug signal because that signal is not defined at this level.